website/src/docs/hotchocolate/v14/fetching-data/dataloader.md

```diff
@@ -255,13 +313,179 @@ public Task<IEnumerable<Person>> GetPersonByLastName(
+
+With a stateful DateLoader you can pass on a selection to the DataLoader which is translated into a expression (`LambdaExpression<Func<TValue, TProjection>>`). Within your DataLoader inject the `ISelectorBuilder` and apply
+it to your `IQueryable<T>`.
+
+```csharp
+internal static class ProductDataLoader
+{
+    [DataLoader]
+    public static async Task<Dictionary<int, Product>> GetProductByIdAsync(
+        IReadOnlyList<int> ids,
+        ISelectorBuilder selector, // selector builder
+        CatalogContext context,
+        CancellationToken ct)
+        => await context.Products
+            .AsNoTracking()
+            .Where(t => ids.Contains(t.Id))
+            .Select(selector, t => t.Id) // apply selector
+            .ToDictionaryAsync(t => t.Id, ct);
+}
+```
+
+In order to apply the selector we provide an extension method called `Select` which applies the `selector` in addition to the key selector. Since the required data might not contain the DataLoader key we have to always provide a key selector as well.
+
+This `ProductByIdDataLoader` is no projectable but will only apply projections if at least one selection passed in from the usage side.
+
+If we would use the `ProductByIdDataLoader` without providing a selection it would just return the full entity.
+
+```csharp
+public class Query
+{
+    public async Task<Product> GetProductAsync(
+        int id,
+        IProductByIdDataLoader productById,
+        CancellationToken cancellationToken)
+        => await productById.LoadAsync(id, cancellationToken);
+}
+```
+
+However if we provide a selection the DataLoader is branched and will return an entity with only the selected fields.
+
+```csharp
+public class Query
+{
+    public async Task<Product> GetProductAsync(
+        int id,
+        IProductByIdDataLoader productById,
+        ISelection selection,
+        CancellationToken cancellationToken)
+        => await productById
+            .Select(selection)
+            .LoadAsync(id, cancellationToken);
+}
+```
+
+Important to note here is that when using projections we can no longer make use of the cache in the same way as before. When we branch a DataLoader we assign the DataLoader a different partition of the cache. This means that only resolvers will share the same cache partition if their selection is translated into the exact same selector expression.
+
+In addition to the GraphQL selection we can also chain in manual includes to our DataLoader call.
+
+```csharp
+public class Query
+{
+    public async Task<Product> GetProductAsync(
+        int id,
+        IProductByIdDataLoader productById,
+        ISelection selection,
+        CancellationToken cancellationToken)
+        => await productById
+            .Select(selection)
+            .Include(t => t.Name)
```

Comment:
`Name` sounds like a property, not a related entity?

Reply:
Thats correct ... you can also include like normal fields